app/frontend/controllers/tab_controller.js

```diff
@@ -422,6 +437,38 @@ class TabController {
     uiHelper.configureButtonStyles('.ui-tabs-nav');
   }
 
+  saveTabScrollPosition(tabIndex) {
+    var metaTab = this.getTab(tabIndex);
+    var firstVerseListAnchor = uiHelper.getFirstVisibleVerseAnchor();
+
+    if (metaTab != null && firstVerseListAnchor != null) {
+      metaTab.setLocation(firstVerseListAnchor);
+    }
+  }
+
+  savePreviousTabScrollPosition() {
+    if (this.lastSelectedTabIndex != null) {
+      this.saveTabScrollPosition(this.lastSelectedTabIndex);
+    }
+  }
+
+  restoreScrollPosition(tabIndex) {
+    var metaTab = this.getTab(tabIndex);
+
+    if (metaTab != null) {
+      var currentVerseListFrame = app_controller.getCurrentVerseListFrame(tabIndex);
+
+      if (currentVerseListFrame != null) {
+        var savedScrollTop = metaTab.getLocation();
```

Comment:
Maybe `const savedScrollTop` would be better

Reply:
Yes!